### PR TITLE
Add yogar-cbmc-parallel.py

### PR DESCRIPTION
**Pull request by yinliangze**

### CODE REVIEW

File: benchexec/tools/yogar-cbmc-parallel.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        elif 'FAILED' in stroutput:
+            status = result.RESULT_FALSE_REACH
+        elif 'SUCCESSFUL' in stroutput:
+            status = result.RESULT_TRUE_PROP
+        elif 'UNKNOWN' in stroutput:
+            status = result.RESULT_UNKNOWN
+        else:
+            status = result.RESULT_UNKNOWN
+
+return status
```
Comment (PhilippWendler) [inline benchexec/tools/yogar-cbmc-parallel.py]: This wrong indentation needs to be fixed.

**Comment by yinliangze** [comment]
Thanks very much. I have fixed the indentation problem and submitted a new commit. We do not have a version number yet.

File: benchexec/tools/yogar-cbmc-parallel.py (hunk 1 above)
Comment (yinliangze) [reply benchexec/tools/yogar-cbmc-parallel.py]: Thanks very much. We have fixed the indentation problem and submitted a new commit. We do not have a version number yet.

[the remark this answers is not in the document]
Comment (yinliangze) [reply benchexec/tools/yogar-cbmc-parallel.py]: Thanks very much. We have fixed the indentation problem and submitted a new commit.

**Comment by PhilippWendler** [comment]
Thanks. How about the inheritance from the other tool-info module for Yogar-CBMC?

Note that you don't even need another tool-info module if the parallel version is just another configuration of Yogar-CBMC. You can have several submissions to SV-COMP with the same tool-info module.

**Comment by PhilippWendler** [comment]
Please remove the erroneous commit 3eaceab (by force-pushing to your branch).

**Comment by yinliangze** [comment]
Dear Philipp,


We have fixed the identation problem of yogar-cbmc-parallel and submitted a new commit. Can you see that on the github? We found that it appears some times and when we refresh the page later it disappears then.


Best,


Liangze Yin





--


Liangze Yin
School of Computer,
National University of Defense Technology,
Changsha, 410074, P.R.China,
yinliangze@163.com



At 2018-10-21 21:58:45, "Philipp Wendler" <notifications@github.com> wrote:


@PhilippWendler requested changes on this pull request.

This looks like a clone of the tool-info module of Yogar-CBMC. If Yogar-CBMC-Parallel is just a variant of Yogar-CBMC, then it should probably reuse the code of yogar-cbmc.py by inheriting from the tool-info module there.

Does Yogar-CBMC-Parallel have a version number? If yes, please add a version method that reads the version number from the tool and returns it.

In benchexec/tools/yogar-cbmc-parallel.py:

> +
+        status = result.RESULT_UNKNOWN
+        stroutput = str(output)
+
+        if isTimeout:
+            status = 'TIMEOUT'
+        elif 'FAILED' in stroutput:
+            status = result.RESULT_FALSE_REACH
+        elif 'SUCCESSFUL' in stroutput:
+            status = result.RESULT_TRUE_PROP
+        elif 'UNKNOWN' in stroutput:
+            status = result.RESULT_UNKNOWN
+        else:
+            status = result.RESULT_UNKNOWN
+
+return status


This wrong indentation needs to be fixed.

**Comment by yinliangze** [comment]
It is another tool using yogar-cbmc, not another configuration of Yogar-CBMC. Hence, the tool name and the options are all different. I still cannot understand how to remove the commit.  I just need to push the commit to my branch?


Best,


Liangze Yin




--


Liangze Yin
School of Computer,
National University of Defense Technology,
Changsha, 410074, P.R.China,
yinliangze@163.com



At 2018-10-22 23:28:34, "Philipp Wendler" <notifications@github.com> wrote:


Thanks. How about the inheritance from the other tool-info module for Yogar-CBMC?

Note that you don't even need another tool-info module if the parallel version is just another configuration of Yogar-CBMC. You can have several submissions to SV-COMP with the same tool-info module.

**Comment by PhilippWendler** [comment]
In that case, another tool-info module is fine, but it should still use inheritance instead of code duplication.

For removing the commit, you need to adjust your branch locally to not contain the commit (using git rebase, or git reset --hard, or similar), and then force push to GitHub.

**Comment by yinliangze** [comment]
Dear Philipp,



I have removed the 3eacea commit of my branch using the git reset --hard command. We also tried to inherit yogar-cbmc.py. However, it seems that the name of "yogar-cbmc.py" cannot contain the symbol '-' if it need to be inherited. Hence, to do that, we need to first modify the name of "yogar-cbmc.py" into some other name like "yogar_cbmc.py". What  should we do right now?


Best,


Liangze Yin



--


Liangze Yin
School of Computer,
National University of Defense Technology,
Changsha, 410074, P.R.China,
yinliangze@163.com



At 2018-10-23 12:46:23, "Philipp Wendler" <notifications@github.com> wrote:


In that case, another tool-info module is fine, but it should still use inheritance instead of code duplication.

For removing the commit, you need to adjust your branch locally to not contain the commit (using git rebase, or git reset --hard, or similar), and then force push to GitHub.

**Comment by PhilippWendler** [comment]
Ah yes. I wouldn't like to rename `yogar-cbmc.py` because the name was used already, but there is a workaround:

```
# Workaround for importing yogar-cbmc despite the dash
yogar_cbmc = __import__("benchexec.tools.yogar-cbmc", fromlist=["Tool"])

class Tool(yogar_cbmc.Tool):
    ....
````

**Comment by yinliangze** [comment]
Dear Philipp,



We have modified the yogar-cbmc-parallel.py file. Is it ok now? Thanks very much for your help.


Best,


Liangze Yin





--


Liangze Yin
School of Computer,
National University of Defense Technology,
Changsha, 410074, P.R.China,
yinliangze@163.com



At 2018-10-24 12:36:28, "Philipp Wendler" <notifications@github.com> wrote:


Ah yes. I wouldn't like to rename yogar-cbmc.py because the name was used already, but there is a workaround:

# Workaround for importing yogar-cbmc despite the dash
yogar_cbmc = __import__("benchexec.tools.yogar-cbmc", fromlist=["Tool"])

class Tool(yogar_cbmc.Tool):
    ....

File: benchexec/tools/yogar-cbmc-parallel.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class Tool(yogar_cbmc.Tool):
+
+    def executable(self):
+        return util.find_executable('yogar-cbmc-parallel')
+
+    def name(self):
+        return 'Yogar-CBMC-Parallel'
+
+    def cmdline(self, executable, options, tasks, propertyfile, rlimits):
+        return [executable] + tasks
```
Comment (PhilippWendler) [inline benchexec/tools/yogar-cbmc-parallel.py]: Is there a specific reason why you ignore `options` here? This means that all command-line parameters specified in `<option>` tags in the benchmark XML will be silently ignored.

**Comment by yinliangze** [comment]
No. We have added it in the new commit.



--


Liangze Yin
School of Computer,
National University of Defense Technology,
Changsha, 410074, P.R.China,
yinliangze@163.com



At 2018-10-24 16:11:47, "Philipp Wendler" <notifications@github.com> wrote:


@PhilippWendler commented on this pull request.

In benchexec/tools/yogar-cbmc-parallel.py:

> +limitations under the License.
+"""
+
+import benchexec.util as util
+yogar_cbmc = __import__("benchexec.tools.yogar-cbmc", fromlist=["Tool"])
+
+class Tool(yogar_cbmc.Tool):
+
+    def executable(self):
+        return util.find_executable('yogar-cbmc-parallel')
+
+    def name(self):
+        return 'Yogar-CBMC-Parallel'
+
+    def cmdline(self, executable, options, tasks, propertyfile, rlimits):
+        return [executable] + tasks


Is there a specific reason why you ignore options here? This means that all command-line parameters specified in <option> tags in the benchmark XML will be silently ignored.

**Pull request merged**